File: Extensions.Tests/AsStringTests.fs
namespace Extensions

open System
open System.Text
open Stringier
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type AsStringTests() =
    [<TestMethod>]
    member _.``runes AsString - hello`` () =
        Assert.AreEqual("hello", [|Rune(0x68); Rune(0x65); Rune(0x6C); Rune(0x6C); Rune(0x6F)|].AsString())

    [<TestMethod>]
    member _.``runes AsString - привет`` () =
        Assert.AreEqual("привет", [|Rune(0x43F); Rune(0x440); Rune(0x438); Rune(0x432); Rune(0x435); Rune(0x442)|].AsString())

    [<TestMethod>]
    member _.``runes AsString - 𝄞`` () =
        Assert.AreEqual("𝄞", [|Rune(0x1D11E)|].AsString())

File: Tests/AsStringTests.cs
using System;
using System.Text;
using Stringier;
using Xunit;

namespace Tests {
	public class AsStringTests {
		[Theory]
		[InlineData(new[] { 'h', 'e', 'l', 'l', 'o' }, "hello")]
		public void AsString_Chars(Char[] chars, String expected) {
			Assert.Equal(expected, chars.AsString());
			Assert.Equal(expected, chars.AsSpan().AsString());
		}

		[Theory]
		[InlineData(new[] { 0x68, 0x65, 0x6C, 0x6C, 0x6F }, "hello")]
		[InlineData(new[] { 0x43F, 0x440, 0x438, 0x432, 0x435, 0x442 }, "привет")]
		[InlineData(new[] { 0x1D11E }, "𝄞")]
		public void AsString_Runes(Int32[] scalarValues, String expected) {
			Rune[] runes = new Rune[scalarValues.Length];
			for (Int32 i = 0; i < runes.Length; i++) {
				runes[i] = new Rune(scalarValues[i]);
			}
			Assert.Equal(expected, runes.AsString());
			Assert.Equal(expected, runes.AsSpan().AsString());
		}
	}
}
